src/project.rs

```diff
@@ -33,10 +41,18 @@ pub struct FileInfo {
 
 impl FileInfo {
     pub fn from_path(path: &Path) -> Result<FileInfo, io::Error> {
+        println!("GETTING ONE FILE NOW {:?}", path);
         let file = File::open(path)?;
+        // more dumbass heuristics to avoid scanning binaries
+        println!("len {}", file.metadata().unwrap().len());
+        if file.metadata().unwrap().len() > 5000000 {
+            return Ok(FileInfo::blank())
+        }
```

Comment:
I wonder if we can try to scan the first chunk of the file and see if it is utf-8 or something and if not, then bail.

Reply:
This is dumb but it's a crazy hard problem.

For ELF executables, we can use an existing parser library. nrc wrote one https://github.com/nrc/xmas-elf

Reply:
Interesting...